object/swift.go

```diff
@@ -0,0 +1,170 @@
+package object
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"github.com/ncw/swift"
+)
+
+type SwiftOSS struct {
+	conn		*swift.Connection
+	region		string
+	storageUrl	string
+	container 	string
+}
+
+func (s *SwiftOSS) String() string {
+	return fmt.Sprintf("region:[%s], %s/%s", s.region, s.storageUrl, s.container)
+}
+
+func (s *SwiftOSS) Create() error {
+	var err error
+	err = s.conn.ContainerCreate(s.container, nil)
+	if err != nil {
+		fmt.Printf("ContainerCreate error: %v", err)
+	}
+	return err
+}
+
+func (s *SwiftOSS) Head(key string) (*Object, error) {
+	info, _, err := s.conn.Object(s.container, key)
+	if err != nil {
+		return nil, err
+	}
+
+	mtime := info.LastModified
+	size := info.Bytes
+
+	return &Object{
+		key,
+		size,
+		mtime,
+		info.PseudoDirectory,
+	}, nil
+}
+
+func (s *SwiftOSS) Get(key string, off, limit int64) (io.ReadCloser, error) {
+	objOpenFile, _, err := s.conn.ObjectOpen(s.container, key, false, nil)
+	if err != nil {
+		fmt.Printf("ObjectOpen error: %v", err)
+		return nil, err
+	}
+	if off > 0 {
+		_, err := objOpenFile.Seek(off, 0)
+		if err != nil {
+			objOpenFile.Close()
+			fmt.Printf("object seek error: %v", err)
+			return nil, err
+		}
+	}
+	if limit > 0 {
+		defer objOpenFile.Close()
+		buf := make([]byte, limit)
+		if n, err := objOpenFile.Read(buf); err != nil {
+			return nil, err
+		} else {
+			return ioutil.NopCloser(bytes.NewBuffer(buf[:n])), nil
+		}
+	}
+	return objOpenFile, err
+}
+
+func (s *SwiftOSS) Put(key string, in io.Reader) error {
+	_, err := s.conn.ObjectPut(s.container, key, in, false, "", "", nil)
+	return err
+}
+
+func (s *SwiftOSS) Copy(dst, src string) error {
+	return notSupported
+}
+
+func (s *SwiftOSS) Delete(key string) error {
+	err := s.conn.ObjectDelete(s.container, key)
+	return err
+}
+
+func (s *SwiftOSS) List(prefix, marker string, limit int64) ([]*Object, error) {
+	return nil, notSupported
+}
+
+func (s *SwiftOSS) ListAll(prefix, marker string) (<-chan *Object, error) {
```

Comment:
Since we don't implement ListAll() or List(), the it can' be used with `juicesync`, should we put these into `juicefs`?